lib/ts/wrappers/express.ts

```diff
@@ -0,0 +1,268 @@
+
+export class ExpressRequest extends BaseRequest {
+    private request: Request;
+    private parserChecked: boolean;
+
+    constructor(request: Request) {
+        super();
+        this.request = request;
+        this.parserChecked = false;
+    }
+
+    getKeyValueFromQuery = async (key: string): Promise<string | undefined> => {
+        if (!this.parserChecked) {
+            await assertThatBodyParserHasBeenUsedForExpressLikeRequest(this.getMethod(), this.request);
+            this.parserChecked = true;
+        }
+        if (this.request.query === undefined) {
+            return undefined;
+        }
+        let value = this.request.query[key];
+        if (value === undefined || typeof value !== "string") {
+            return undefined;
+        }
+        return value;
+    };
+
+    getJSONBody = async (): Promise<any> => {
+        if (!this.parserChecked) {
+            await assertThatBodyParserHasBeenUsedForExpressLikeRequest(this.getMethod(), this.request);
+            this.parserChecked = true;
+        }
+        return this.request.body;
+    };
+
+    getMethod = (): HTTPMethod => {
+        return normaliseHttpMethod(this.request.method);
+    };
+
+    getCookieValue = (key: string): string | undefined => {
+        return getCookieValueFromIncomingMessage(this.request, key);
+    };
+
+    getHeaderValue = (key: string): string | undefined => {
+        return getHeaderValueFromIncomingMessage(this.request, key);
+    };
+
+    getOriginalURL = (): string => {
+        return this.request.originalUrl;
+    };
+}
+
+export class ExpressResponse extends BaseResponse {
+    private response: Response;
+    private statusCode: number;
+
+    constructor(response: Response) {
+        super();
+        this.response = response;
+        this.statusCode = 200;
+    }
+
+    setHeader = (key: string, value: string, allowDuplicateKey: boolean) => {
+        setHeaderForExpressLikeResponse(this.response, key, value, allowDuplicateKey);
+    };
+
+    setCookie = (
+        key: string,
+        value: string,
+        domain: string | undefined,
+        secure: boolean,
+        httpOnly: boolean,
+        expires: number,
+        path: string,
+        sameSite: "strict" | "lax" | "none"
+    ) => {
+        setCookieForServerResponse(this.response, key, value, domain, secure, httpOnly, expires, path, sameSite);
+    };
+
+    /**
+     * @param {number} statusCode
+     */
+    setStatusCode = (statusCode: number) => {
+        this.statusCode = statusCode;
+    };
+
+    sendJSONResponse = (content: any) => {
+        if (!this.response.writableEnded) {
+            this.response.status(this.statusCode).json(content);
+        }
+    };
+}
+
+export interface SessionRequest extends Request {
+    session?: SessionContainerInterface;
+}
+
+const ExpressWrapper: Wrapper = {
+    middleware: () => {
+        return async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
I do think a lot of this function can be moved into `supertokens.ts`. Just like you are assuming for APIs that the signature is `(req: BaseRequest, res: BaseResponse) => Promise<boolean>`, you can essentially assume the same signature for this function in `supertokens.ts`. Then your impl here would be something like:
```
return async (req: Request, res: Response, next: NextFunction) => {
   let request = new ExpressRequest(req);
   let response = new ExpressResponse(res);
   let requestHandled = await supertokens.middleware(request, response);
   if (!requestHandled) {
      next();
   }
}
```

Reply:
👍🏼